plugins/groovy/groovy-psi/src/org/jetbrains/plugins/groovy/codeInspection/assignment/GroovyResultOfAssignmentUsedInspection.java

```diff
@@ -56,9 +58,13 @@ public BaseInspectionVisitor buildVisitor() {
     @Override
     public void visitAssignmentExpression(GrAssignmentExpression grAssignmentExpression) {
       super.visitAssignmentExpression(grAssignmentExpression);
-      if (PsiUtil.isExpressionUsed(grAssignmentExpression)) {
+      if (isConfusingAssignmentUsage(grAssignmentExpression)) {
         registerError(grAssignmentExpression);
       }
     }
+
+    private static boolean isConfusingAssignmentUsage(PsiElement expr) {
+      return !(expr.getParent() instanceof GrControlFlowOwner) && PsiUtil.isExpressionUsed(expr);
```

Comment:
this is the actual fix, i.e. if it's in the root of the Groovy file, don't mark it.

Common example is the `Settings.gradle` file:
```Groovy
rootProject.name = "spock-example"
```